recipe/migrations/numpy2.yaml

```diff
@@ -49,21 +49,35 @@ __migrator:
     case of git conflicts by marking it as a draft).
 
   migration_number: 1
-  paused: true
   ordering:
     # prefer channels including numpy_rc (otherwise smithy doesn't
     # know which of the two values should be taken on merge)
     channel_sources:
       - conda-forge
       - conda-forge/label/numpy_rc,conda-forge
 
-# needs to match length of zip {python, python_impl, numpy}
-# as it is in global CBC in order to override it
+# we need to override the entire zip {python, python_impl, numpy}
+# in order to change the length of it; that's the only way to migrate
+# python 3.12 for numpy 2.0, as we cannot update the existing 3.12
+# migrator without breaking already-migrated feedstocks immediately.
 numpy:
   - 1.22  # no py38 support for numpy 2.0
   - 2.0
```

Comment:
I don't think this is relevant for conda-forge, where we can repopatch existing builds to have a `<2.0` constraint (see https://github.com/conda-forge/conda-forge-repodata-patches-feedstock/pull/712). So old scikit-image builds would never break by having py3.9 builds for numpy 2.0.

Reply:
lets follow numpy upstream for their supported workflows.

Ultimately, I want to stop us from "creating all the patches" for other projects to support numpy 2.0 + python 3.9.

Our job should "end" at one point in time.

Reply:
Ultimately, I want to respect and the to help the numpy developers in this transition.

If our goal is to proactively transition packages and report bugs upstream, we should do that, we shouldn't be creating more testing surface, and breakage, than is necessary.

I strongly believe that if numpy upstream developers decide to end support for an old version of python for numpy 2, we should honor that decision.

If they want to keep it, we should use our ability to help identify breakages ahead of a wider spread release.

Reply:
> lets follow numpy upstream for their supported workflows.

Which you're proposing to change last minute. Of course, we should follow that for numpy 2.0 GA, but it doesn't have any bearing IMO on the rc-migration.

Reply:
Isn't the whole point of the rc-migration to stay ahead of the large re-compilation influx and to better predict breakages? 

Why would we increase the testing surface for many of the upstream developers (many numpy folk here and the rest of the open source science stack)?

Real concerns against widestream patching were brought up https://github.com/conda-forge/conda-forge-repodata-patches-feedstock/pull/712

We should be weary of wide scale patching and intrusive behavior. Again, I'm trying to honor upstream's wishes here, and I've made my point to them. We should wait a day or two for them to decide.

Reply:
> Isn't the whole point of the rc-migration to stay ahead of the large re-compilation influx and to better predict breakages?

I'd say "prepare as many feedstocks as possible for a post-2.0 world before GA", but that's pretty much the same.

> Why would we increase the testing surface for many of the upstream developers?

I think that's red herring. In 99% of the cases, a package will either build or not build with numpy 2.0. For projects still supporting 3.9, I strongly doubt that this will create additional failure modes compared to those existing for 3.10+.

Also, this only concerns the rc-migration; I've already said we should match what numpy 2.0 GA does when we start publishing to the main channel (e.g. we could simply never publish the 3.9 build of numpy 2.0 there).

> Real concerns against widestream patching were brought up [conda-forge/conda-forge-repodata-patches-feedstock#712](https://github.com/conda-forge/conda-forge-repodata-patches-feedstock/pull/712)

Sure, and they can be addressed (which @beckermr offered to do).

> Again, I'm trying to honor upstream's wishes here

That's a bit of a stretch IMO. They're your wishes, which upstream might choose to adopt, AFAICT mainly because PyPI doesn't have a way to do retroactive constraint patches.

> We should wait a day or two for them to decide.

It's fine to wait a day or two, sure. What I don't want is to hold up the migration for concerns that really only matter for numpy 2.0 GA. If we build a couple extra builds during the rc-migration for 3.9, that's completely harmless (unless for builds failing specifically on 3.9, who can then skip that), but if we migrate without 3.9 we cannot go back without a lot of manual effort if upstream decides to keep 3.9 support after all.

Reply:
Thanks for waiting a day or two, i know it isn't desirable when things are so close after such a long time.